### PR TITLE
Fixed qrcode regex analysis to read the parameters (amount, vendorFie…

### PR DESCRIPTION
**Pull request by alucqrd**
## Proposed changes
Fixed qrcode regex analysis to read the parameters (amount, vendorField and label).
If user is scanning a qrcode while creating a new contact, field label is used for the name of the contact (scheme must be: ark:ARK_ADDRESS?label=CONTACT_NAME).

If user is scanning a qrcode while creating a new contact, field label is used for the name of the contact (scheme must be: ark:ARK_ADDRESS?label=CONTACT_NAME).

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/components/qr-scanner/qr-scanner.ts [hunk 1]
```diff
@@ -41,6 +41,21 @@ export class QRScannerComponent {
 
     if (qrCode.match(prefixUriRegex)) {
       scheme.address = prefixUriRegex.exec(qrCode)[1];
+      const paramsRegex = new RegExp(`(\\?|\\&)([^=]+)\\=([^&]+)`, 'g');
+      let regexResult;
+      while ((regexResult = paramsRegex.exec(qrCode)) != null) {
```
Comment (alexbarnsley) [inline src/components/qr-scanner/qr-scanner.ts]: I'm not sure how this loop can ever change? `qrCode` never changes, which would surely result in an infinite loop if `regexResult != null`?

Comment (alucqrd) [reply src/components/qr-scanner/qr-scanner.ts]: paramsRegex is updated each time exec is runned.

"If the match succeeds, the exec() method returns an array and updates properties of the regular expression object. The returned array has the matched text as the first item, and then one item for each capturing parenthesis that matched containing the text that was captured."

source : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec

Comment (alucqrd) [reply src/components/qr-scanner/qr-scanner.ts]: Any news / need for some rework ?

Comment (faustbrian) [reply src/components/qr-scanner/qr-scanner.ts]: @alexbarnsley

**Pull request merged**